File: lab/examples/go/internal/hexagon/application/order/createorder/ports/id_generator.go
// Package ports holds contracts that only the create-order slice needs.
//
// IDGenerator is a use-case-local port: it lives inside the slice because no
// other use case requires it. This is the "as local as possible" level of a
// HexSlice port.
package ports

import "hexslice/example/internal/hexagon/domain/order"

// IDGenerator produces new, unique order identifiers.
type IDGenerator interface {
	NewOrderID() order.ID
}

File: lab/examples/kotlin/hexagon/application/src/main/kotlin/hexslice/hexagon/application/order/createorder/ports/outbound/IdGenerator.kt
// The outbound (driven) port that only the create-order slice needs.
//
// It lives inside the slice because no other use case requires it: the "as local
// as possible" level of a HexSlice port.
package hexslice.hexagon.application.order.createorder.ports.outbound

import hexslice.hexagon.domain.order.OrderId

/** Produces new, unique order identifiers. */
interface IdGenerator {
    fun newOrderId(): OrderId
}
